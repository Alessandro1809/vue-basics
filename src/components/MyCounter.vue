<script lang="ts" setup>//setup script
import {useCounter} from '../composables/useCounter';

   
   
    interface Props {
        value: number;
    }
    const props = defineProps<Props>();
    
    
    const {counter,squareCounter,incrementCounter,decremmentCounter} = useCounter();
</script>

<template>
    <section>
        <h3>Counter : {{counter}}</h3>
        <h3>Square : {{squareCounter}}</h3>

        <div class="flex gap-2">
            <button
            class="btn"
            @click="incrementCounter">+1</button>
            <button 
            class="btn"
            @click="decremmentCounter">-1</button>
        </div>
    </section>

</template>



//scoped style
<style scoped>

.btn{
    @apply bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded;
}

</style>