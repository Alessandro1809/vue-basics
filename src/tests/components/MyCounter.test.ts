import { describe, expect, test, it } from 'vitest';
import MyCounter from '../../components/MyCounter.vue';
import { mount } from '@vue/test-utils';

describe('MyCounter', () => {
  test('should increment the counter', () => {});

  test('should match the snapshot', () => {
    const wrapper = mount(MyCounter, {
      props: {
        value: 5,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Render the counter value correctly', () => {
    //preparar la prueba
    const value = 5;
    //estimular el componente
    const wrapper = mount(MyCounter, {
      props: {
        value: value,
      },
    });
    const [counterLabel, squareLabel] = wrapper.findAll('h3');
    //comprobar el resultado
    expect(wrapper.find('h3').text()).toContain(`Counter : ${value}`);
    expect(wrapper.find('[data-test-id="square-test-label"]').text()).toContain(
      `Square : ${value * value}`,
    );

    expect(counterLabel.text()).toContain(`Counter : ${value}`);
    expect(squareLabel.text()).toContain(`Square : ${value * value}`);
  });

  test('increments the counter value when the buttonn is clicked', async () => {
    //preparar la prueba
    const value = 5;
    //estimular el componente
    const wrapper = mount(MyCounter, {
      props: {
        value: value,
      },
    });
    const [counterLabel, squareLabel] = wrapper.findAll('h3');

    const btnIncremment = wrapper.find('button');
    await btnIncremment.trigger('click');

    expect(counterLabel.text()).toContain(`Counter : ${value + 1}`);
    expect(squareLabel.text()).toContain(`Square : ${(value + 1) * (value + 1)}`);
  });

  test('decrements the counter value when the buttonn is clicked', async () => {
    //preparar la prueba
    const value = 5;
    //estimular el componente
    const wrapper = mount(MyCounter, {
      props: {
        value: value,
      },
    });
    const [counterLabel, squareLabel] = wrapper.findAll('h3');

    const btnDecrement = wrapper.findAll('button')[1];

    await btnDecrement.trigger('click');
    await btnDecrement.trigger('click');
    expect(counterLabel.text()).toContain(`Counter : ${value - 2}`);
    expect(squareLabel.text()).toContain(`Square : ${(value - 2) * (value - 2)}`);
  });
});
