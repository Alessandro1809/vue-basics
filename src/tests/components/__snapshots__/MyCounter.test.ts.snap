// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyCounter > should match the snapshot 1`] = `
"<section data-v-0352012b="">
  <h3 data-v-0352012b="">Counter : 5</h3>
  <h3 data-v-0352012b="" data-test-id="square-test-label">Square : 25</h3>
  <div data-v-0352012b="" class="flex gap-2"><button data-v-0352012b="" class="btn">+1</button><button data-v-0352012b="" class="btn">-1</button></div>
</section>"
`;
